src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
+//  instruction where "_emitter" is located.
+//
+// Arguments:
+//  _emitter - an emitter* instance located at the first instruction from
+//   this variable becomes dead.
+//
+// Assumptions:
+//  This variable is becoming dead so it should be alive.
+//
+// Notes:
+//  The position of "_emitter" matters to ensure intervals inclusive of the
+//  beginning and exclusive of the end.
+//
+void VariableLiveDescriptor::endLiveRangeAtEmitter(emitter* _emitter) const
+{
+    noway_assert(hasVariableLiveRangeOpen());
+
+    // Using [close, open) ranges so as to not compute the size of the last instruction
+    m_VariableLiveRanges->back().m_EndEmitLocation.CaptureLocation(_emitter);
+
+    // No m_EndEmitLocation has to be Valid
+    noway_assert(m_VariableLiveRanges->back().m_EndEmitLocation.Valid());
+}
+
+//------------------------------------------------------------------------
+// UpdateLiveRangeAtEmitter: Report this variable as changing its variable
+//  home to "varHome" since the instruction where "_emitter" is located.
+//
+// Arguments:
+//  varHome  - the new variable home.
+//  _emitter - an emitter* instance located at the first instruction from
+//   where "varHome" becomes valid.
+//
+// Assumptions:
+//  This variable is being born so it should be dead.
+//
+// Notes:
+//  The position of "_emitter" matters to ensure intervals inclusive of the
+//  beginning and exclusive of the end.
+//
+void VariableLiveDescriptor::updateLiveRangeAtEmitter(CodeGenInterface::siVarLoc varHome, emitter* _emitter) const
+{
+    // This variable is changing home so it has been started before during this block
+    noway_assert(m_VariableLiveRanges != nullptr && !m_VariableLiveRanges->empty());
+
+    // And its last m_EndEmitLocation has to be invalid
+    noway_assert(!m_VariableLiveRanges->back().m_EndEmitLocation.Valid());
+
+    // If we are reporting again the same home, that means we are doing something twice?
+    // noway_assert(! CodeGenInterface::siVarLoc::Equals(&m_VariableLiveRanges->back().m_VarHome, varLocation));
+
+    // Close previous live range
+    endLiveRangeAtEmitter(_emitter);
+
+    // Open new live range with invalid end
+    m_VariableLiveRanges->emplace_back(varHome, emitLocation(), emitLocation());
+    m_VariableLiveRanges->back().m_StartEmitLocation.CaptureLocation(_emitter);
+
+    // startEmitLocationendEmitLocation has to be Valid and endEmitLocationendEmitLocation  not
+    noway_assert(m_VariableLiveRanges->back().m_StartEmitLocation.Valid());
+    noway_assert(!m_VariableLiveRanges->back().m_EndEmitLocation.Valid());
+}
+
+#ifdef DEBUG
+void VariableLiveDescriptor::dumpAllRegisterLiveRangesForBlock(emitter* _emitter, const CodeGenInterface* codeGen) const
+{
+    // "m_VariableLiveRanges" should has been initialized
+    noway_assert(m_VariableLiveRanges != nullptr);
+
+    if (m_VariableLiveRanges->empty())
+    {
+        printf("None history\n");
```

Comment:
nit: None => No

Reply:
Seems like unnecesary code to begin with. If it's empty then the else code will just print `[]`. For debug code that should be enough.